### PR TITLE
Some bugfixes, and some refactoring

### PR DESCRIPTION
**Pull request by Jack-McDowell**

### CODE REVIEW

File: BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp [hunk 1]
```diff
@@ -1,294 +1,298 @@
...
+
+        if(GetLastError() != ERROR_NO_MORE_ITEMS) {
+            LOG_VERBOSE(1, "EventLogs::ProcessResults: EvtNext failed with " << GetLastError());
+        }
+
+        return results;
+    }
+
+    std::optional<EventLogItem> EventToEventLogItem(const EventWrapper& hEvent,
+                                                    const std::vector<std::wstring>& params) {
```
Comment (ION28) [inline BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp:126]: Codacy found an issue: [Variable 'status' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5931486)

File: BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp [hunk 2]
```diff
@@ -1,294 +1,298 @@
...
+                                                &dwBufferUsed);
+                } else {
+                    if(pProperty)
+                        free(pProperty);
+
+                    LOG_ERROR(L"EventLogs::IsChannelOpen: realloc failed for channel " + channel);
+                    return false;
+                }
+            }
+
```
Comment (ION28) [inline BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp:254]: Codacy found an issue: [Variable 'status' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5931486)

File: BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp [hunk 3]
```diff
@@ -1,294 +1,298 @@
...
+            free(pProperty);
+
+        return pProperty->BooleanVal;
+    }
+
+    bool OpenChannel(const std::wstring& channel) {
+        EVT_HANDLE hChannel = NULL;
+        DWORD status = ERROR_SUCCESS;
+        EVT_VARIANT ChannelProperty;
+        DWORD dwBufferSize = sizeof(EVT_VARIANT);
```
Comment (ION28) [inline BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp:271]: Codacy found an issue: [Variable 'dwBufferUsed' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5931486)

**Comment by Jack-McDowell** [comment]
Addresses  #360

File: BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp [hunk 4]
```diff
@@ -1,294 +1,295 @@
...
+                                                &dwBufferUsed);
+                } else {
+                    if(pProperty)
+                        free(pProperty);
+
+                    LOG_ERROR(L"EventLogs::IsChannelOpen: realloc failed for channel " + channel);
+                    return false;
+                }
+            }
+
```
Comment (ION28) [inline BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp:253]: Codacy found an issue: [Variable 'status' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5931486)

Comment (ION28) [inline BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp:253]: Codacy found an issue: [Variable 'status' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5931486)

**Pull request merged**